src/main/java/com/couchbase/lite/replicator/Replication.java

```diff
@@ -369,6 +377,15 @@ public ReplicationStatus getStatus() {
      */
     @Override
     public void changed(ChangeEvent event) {
+
+        // forget cached IDs (Should be executed in workExecutor)
+        db.runAsync(new AsyncTask() {
+            @Override
+            public void run(Database database) {
+                pendingDocIDs = null;
```

Comment:
I think resetting the `pendingDocIDs` variable here needs a synchronized block as the variable is also accessed on the main thread in the getPendingDocumentIDs() method below.